File: Data Structures & Algorithms/combination-target-sum/submission-1.java
class Solution
 {
    void f(int ind,int[] nums,int target,List<List<Integer>>ans,  List<Integer>list)
    {
        if(target==0)
        {
            // ans.add(list);
               ans.add(new ArrayList<>(list));
            return;
        }
        if(ind>=nums.length || target<0)
        return;
        if(nums[ind]<=target)
            {
                list.add(nums[ind]);
                f(ind,nums,target-nums[ind],ans,list);
                list.remove(list.size()-1);
            }
                f(ind+1,nums,target,ans,list);


    }
    public List<List<Integer>> combinationSum(int[] nums, int target)
     {
        List<List<Integer>>ans=new ArrayList<>();
        List<Integer>list=new ArrayList<>();
        f(0,nums,target,ans,list);
        return ans;
    }
}

File: Data Structures & Algorithms/remove-element/submission-1.py
class Solution:
    def removeElement(self, nums: List[int], val: int) -> int:

        n=len(nums)
        c=0
        ans = []

        for i in range(0,n):
            if(nums[i]!=val):
                ans.append(nums[i])
            else:
                c+=1

        k=n-c
        for i in range(0,k):
            nums[i]=ans[i]

        return k

        